File: 1829-maximum-units-on-a-truck/maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        int count = 0;
        Arrays.sort(boxTypes, (a,b) -> b[1] - a[1]);
        for(int[] a : boxTypes) {
           int min = Math.min(truckSize, a[0]);
           count += min * a[1];
           truckSize -= min;
           if(truckSize == 0) return count;
        }
        return count;
    }
}

File: 1829-maximum-units-on-a-truck/maximum-units-on-a-truck.js
/**
 * @param {number[][]} boxTypes
 * @param {number} truckSize
 * @return {number}
 */
var maximumUnits = function(boxTypes, truckSize) {
    let sorted = boxTypes.sort((a,b) => b[1] - a[1]);
    let count = 0;
    for(let i=0; i< sorted.length;i++){
        if(truckSize > sorted[i][0]){
            count += sorted[i][0] * sorted[i][1];
            truckSize -= sorted[i][0]
        } else if (truckSize > 0){
            count += truckSize * sorted[i][1];
            truckSize = 0;
        } else{
            break;
        }
    }
    return count;
};